nmdc_runtime/api/endpoints/users.py

```diff
@@ -170,3 +182,32 @@ def create_user(
         ).model_dump(exclude_unset=True)
     )
     return mdb.users.find_one({"username": user_in.username})
+
+
+@router.put(
+    "/users",
+    status_code=status.HTTP_200_OK,
+    response_model=User,
+    name="Update User",
+    description="Update information for given username (Admin Only)",
+)
+def update_user(
+    user_in: UserIn,
+    requester: User = Depends(get_current_active_user),
+    mdb: pymongo.database.Database = Depends(get_mongo_db),
+):
+    check_can_create_user(requester)
+    username = user_in.username
+
+    if user_in.password:
+        user_dict = UserInDB(
+            **user_in.model_dump(),
+            hashed_password=get_password_hash(user_in.password),
```

Comment:
I think I will find this comment helpful on future read-throughs.

```suggestion
            # Store the password hash instead of the plain text password.
            hashed_password=get_password_hash(user_in.password),
```